### PR TITLE
Issue #1484 by cableman: Added support for displaying opening hours by category

### PR DESCRIPTION
**Pull request by cableman**
See http://platform.dandigbib.org/issues/1484

### CODE REVIEW

File: themes/ddbasic/gulpfile.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ gulp.task('sass', 'Process SCSS using libsass',
         includePaths: [
           'node_modules/compass-mixins/lib',
           // Zen grids is downloaded as a library using drush make.
-          '../../libraries/zen-grids/stylesheets'
+          './zen-grids/stylesheets'
```
Comment (kasperg) [inline themes/ddbasic/gulpfile.js]: Why this change?

As I see it [zen-grids](https://github.com/ding2/ding2/blob/master/ding2.make#L392) is not placed in the theme folder but as a library.

File: modules/ding_library/ding_library.features.field_instance.inc [hunk 2]
```diff
@@ -753,7 +753,7 @@ function ding_library_field_default_field_instances() {
         'label' => 'above',
         'settings' => array(),
         'type' => 'hidden',
-        'weight' => 13,
+        'weight' => 8,
```
Comment (kasperg) [inline modules/ding_library/ding_library.features.field_instance.inc]: Note to self: These changes of weight have not practical consequence. The order will remain the same.

File: themes/ddbasic/scripts/ddbasic.common.js [hunk 3]
```diff
@@ -4,13 +4,6 @@
    * Toggle opening hours
    */
   function toggle_opening_hours() {
-    // Create toggle link
-    $('<a />', {
-      'class' : 'opening-hours-toggle js-opening-hours-toggle js-collapsed',
-      'href' : Drupal.t('#toggle-opening-hours'),
-      'text' : Drupal.t('Opening hours')
-    }).insertBefore('.js-opening-hours-toggle-element');
-
```
Comment (kasperg) [inline themes/ddbasic/scripts/ddbasic.common.js]: As I see it this moves rendering of the toggle link from JS to PHP (template.php line 283). What is the reason behind this change?

File: themes/ddbasic/template.php [hunk 4]
```diff
@@ -272,20 +272,32 @@ function ddbasic_preprocess_user_picture(&$variables) {
...
-  // Opening hours on library list. but not on the search page.
-  $path = drupal_get_path_alias();
-  if (!(strpos($path, 'search', 0) === 0)) {
-    $hooks = theme_get_registry(FALSE);
-    if (isset($hooks['opening_hours_week']) && $variables['type'] == 'ding_library') {
-      $variables['opening_hours'] = theme('opening_hours_week', array('node' => $variables['node']));
+function ddbasic_node_view_alter(&$build) {
+  foreach (array_keys($build) as $field) {
+
+    if (preg_match('/^opening_hours_week_*(\d+)*/', $field, $matches)) {
```
Comment (kasperg) [inline themes/ddbasic/template.php]: `$matches` is undefined here.

Also I would appreciate a comment explaining why using regular expression matching is necessary here. I would expect the field name to be static? Could it be simplified by checking if the field starts with `opening_hours_week_`?

File: themes/ddbasic/template.php [hunk 5]
```diff
@@ -272,20 +272,32 @@ function ddbasic_preprocess_user_picture(&$variables) {
...
+      $build[$field][0]['#prefix'] = '<a class="opening-hours-toggle js-opening-hours-toggle js-collapsed" href="#toggle-opening-hours">'.$build[$field]['#title'].'</a><div class="libraries-opening-hours js-opening-hours-toggle-element">';
+      $build[$field][0]['#suffix'] = '</div>';
+
+      if (!isset($build['opening_hours'])) {
+        $build['opening_hours'] = array();
+      }
+      $build['opening_hours'][] = $build[$field];
+      unset($build[$field]);
```
Comment (kasperg) [inline themes/ddbasic/template.php]: What is going on here with shuffling the field around in the view?

File: themes/ddbasic/templates/node--ding-library.tpl.php [hunk 6]
```diff
@@ -96,12 +96,13 @@
...
       <?php print render($content); ?>
     </div>
-    <?php if (!empty($opening_hours)) : ?>
-    <div class="libraries-opening-hours js-opening-hours-toggle-element">
-      <?php print $opening_hours;  ?>
-    </div>
+    <?php if (isset($content['opening_hours'])) : ?>
+      <?php foreach ($content['opening_hours'] as $opening_hours) : ?>
+        <?php print render($opening_hours);  ?>
+      <?php endforeach; ?>
```
Comment (kasperg) [inline themes/ddbasic/templates/node--ding-library.tpl.php]: Why is opening hours treated as an array? It does not seem as if we support multiple opening hours.

File: themes/ddbasic/templates/opening-hours/opening_hours_week.tpl.php [hunk 7]
```diff
@@ -8,7 +8,7 @@
   print $preface;
 endif;
 ?>
-<div class="opening-hours-week placeholder" data-nid="<?php print $node->nid; ?>">
+<div class="opening-hours-week placeholder" data-nid="<?php print $node->nid; ?>" data-tid="<?php print $tid; ?>">
```
Comment (kasperg) [inline themes/ddbasic/templates/opening-hours/opening_hours_week.tpl.php]: Why is this change relevant to solving the issue? I do not see anywhere else that `data-tid` is referenced.

File: themes/ddbasic/templates/panels/panels-pane--week.tpl.php [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+ * - $more: An optional 'more' link (destination only)
+ * - $admin_links: Administrative links associated with the content
+ * - $feeds: Any feed icons or associated with the content
+ * - $display: The complete panels display object containing all kinds of
+ *   data including the contexts and all of the other panes being displayed.
+ *
+ * Changed: Add t() function to pane title.
+ */
+?>
+<a class="opening-hours-toggle js-opening-hours-toggle js-collapsed" href="#toggle-opening-hours"><?php print t('Openings hours') ?></a>
```
Comment (kasperg) [inline themes/ddbasic/templates/panels/panels-pane--week.tpl.php]: This entire line seems to be a change?

**Comment by kasperg** [comment]
Review complete. I would appreciate if this build was rebased on master so we get the build to run as well.

**Comment by cableman** [comment]
This is no longer needed as the new theme handles it.
